webapp/src/components/AssetPage/BestBuyingOption/BestBuyingOption.tsx

```diff
@@ -63,16 +69,34 @@ const BestBuyingOption = ({ asset, tableRef, isOffchainPublicNFTOrdersEnabled }:
             if (response.data.length > 0) {
               setBuyOption(BuyOptions.BUY_LISTING)
               setListing({ order: response.data[0], total: response.total })
-              bidAPI
-                .fetchByNFT(asset.contractAddress, response.data[0].tokenId, ListingStatus.OPEN, BidSortBy.MOST_EXPENSIVE, '1')
-                .then(response => {
-                  setIsLoading(false)
-                  setMostExpensiveBid(response.data[0])
-                })
-                .catch(error => {
-                  console.error(error)
-                  setIsLoading(false)
-                })
+              if (isOffchainPublicNFTOrdersEnabled) {
+                marketplaceAPI
+                  .fetchBids({
+                    contractAddress: asset.contractAddress,
+                    tokenId: response.data[0].tokenId,
+                    status: ListingStatus.OPEN,
+                    sortBy: BidSortBy.MOST_EXPENSIVE,
+                    limit: 1
+                  })
+                  .then(({ results }) => {
+                    setIsLoading(false)
+                    setMostExpensiveBid(results[0])
+                  })
+                  .catch(error => {
+                    console.error(error)
```

Comment:
We're missing here the `setIsLoading(false)`, would you mind adding it?
```suggestion
                    console.error(error)
                    setIsLoading(false)
```